### PR TITLE
Fixes #542 ValueOrigin in PK-Sim compound building block

### LINKED ISSUE #542: ValueOrigin in PK-Sim compound building block
see discussion here https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/234

### PR DESCRIPTION
**Pull request by msevestre**
WIP #542 add value origin command to compound

WIP #542. Add value origin presenter that can display the value origin as a single line.

Update logic to use the Default flag in compound

Add some tests and rename some constants variable that were still outdated

Add some tests and rename some constants variable that were still outdated

More work on updating default

Simplify code by removing value origin management in edit parameter command. We have a command for that already!

Prepare for PR

### CODE REVIEW

File: src/PKSim.Core/Commands/EditParameterCommand.cs [hunk 1]
```diff
@@ -9,9 +9,10 @@ namespace PKSim.Core.Commands
    public abstract class EditParameterCommand : BuildingBlockChangeCommand
    {
       protected IParameter _parameter;
-      protected ValueOrigin _oldValueOrigin;
```
Comment (msevestre) [inline src/PKSim.Core/Commands/EditParameterCommand.cs]: This is now gone from the command and is being dealt with from another command. Separation of concerns

File: src/PKSim.Core/Commands/SetParameterDefaultStateCommand.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+using PKSim.Assets;
+
+namespace PKSim.Core.Commands
+{
+   public class SetParameterDefaultStateCommand : EditParameterCommand
+   {
+      private readonly bool _isDefault;
+      private readonly bool _oldIsDefault;
+
+      public SetParameterDefaultStateCommand(IParameter parameter, bool isDefault) : base(parameter)
```
Comment (msevestre) [inline src/PKSim.Core/Commands/SetParameterDefaultStateCommand.cs]: Simple SetParameterDEfaultStateCommand.

File: src/PKSim.Core/Commands/UpdateParametersValueOriginCommand.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility.Extensions;
+
+namespace PKSim.Core.Commands
+{
+   public class UpdateParametersValueOriginCommand : PKSimMacroCommand
```
Comment (msevestre) [inline src/PKSim.Core/Commands/UpdateParametersValueOriginCommand.cs]: This command is updating a bunch of parameters value origin at once

File: src/PKSim.Core/CoreConstants.cs [hunk 4]
```diff
@@ -911,39 +911,37 @@ public static string ParameterCompoundType(int index)
...
-         public const string VolumeFractionWaterIntra_RR = "Vf (intracellular water)-RR";
-         public const string VolumeFractionLipidsScaleFactor = "Volume fraction lipids scale factor";
-         public const string VolumeFractionProteinsScaleFactor = "Volume fraction proteins scale factor";
-         public const string VolumeFractionWaterScaleFactor = "Volume fraction water scale factor";
-         public const string FractionIntracellular = "Fraction intracellular";
-         public const string FractionInterstitial = "Fraction interstitial";
-         public const string FractionVascular = "Fraction vascular";
-         public const string NumberOfParticlesFactor = "Number_Of_Particles_Factor";
-         public const string StartParticleRadius = "Particle radius (at t=0)";
+         public const string UNDEFINED_ONTOGENY = "Undefined";
```
Comment (msevestre) [inline src/PKSim.Core/CoreConstants.cs]: DA ich schon dabei war....habe ich alle constants endlich umbenannt

File: src/PKSim.Core/Services/BuildConfigurationTask.cs [hunk 5]
```diff
@@ -56,6 +59,7 @@ public IBuildConfiguration CreateFor(Simulation simulation, bool shouldValidate,
 
          //STEP1: Create spatial structure
          buildConfiguration.SpatialStructure = _spatialStructureFactory.CreateFor(individual, simulation);
+         _parameterDefaultStateUpdater.UpdateDefaultFor(buildConfiguration.SpatialStructure);
```
Comment (msevestre) [inline src/PKSim.Core/Services/BuildConfigurationTask.cs]: Code is not implemented yet in ParameterDEfaultStateUpdater. But calls are already available

File: src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+{
+   public interface IParameterDefaultStateUpdater
+   {
+      void UpdateDefaultFor(ISpatialStructure spatialStructure);
+      void UpdateDefaultFor(IEventGroupBuildingBlock eventGroupBuildingBlock);
+   }
+
+   public class ParameterDefaultStateUpdater : IParameterDefaultStateUpdater
+   {
+      public void UpdateDefaultFor(ISpatialStructure spatialStructure)
```
Comment (msevestre) [inline src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs]: This will be part of another PR

File: src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      {
+         //TODO implement
+      }
+
```
Comment (Yuri05) [inline src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs]: open TODO

File: src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+            spatialStructureOrganism.Parameter(x).IsDefault = true;
+         });
+      }
+
+
+      public void UpdateDefaultFor(IEventGroupBuildingBlock eventGroupBuildingBlock)
+      {
+         //TODO implement
+      }
+   }
```
Comment (Yuri05) [inline src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs]: one more TODO

Comment (msevestre) [reply src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs]: same here
https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/543

File: src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs (hunk 7 above)
Comment (msevestre) [reply src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs]: yes see other comment. This will be done in another PR
https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/543

File: src/PKSim.Core/Services/BuildConfigurationTask.cs (hunk 5 above)
Comment (msevestre) [reply src/PKSim.Core/Services/BuildConfigurationTask.cs]: that's the comment here.

**Comment by msevestre** [comment]
Merging
